rocky/Dockerfile

```diff
@@ -31,12 +31,19 @@ RUN --mount=type=cache,target=/var/cache/apt \
 # dependencies
 ARG ENVIRONMENT
 
+# This works around https://github.com/pypa/pip/issues/6469 by installing the git
+# requirements separately
 COPY rocky/requirements*.txt .
 RUN --mount=type=cache,target=/root/.cache \
     pip install --upgrade pip \
     && if [ "$ENVIRONMENT" = "dev" ]; \
-    then pip install -r requirements-dev.txt; \
-    else pip install -r requirements.txt; fi
+    then grep -v git+https:// requirements-dev.txt > /tmp/requirements-nogit.txt ; \
+    grep git+https:// requirements-dev.txt > /tmp/requirements-git.txt ; \
+    pip install -r /tmp/requirements-nogit.txt && pip install -r /tmp/requirements-git.txt ; \
+    else grep -v git+https:// requirements.txt > /tmp/requirements-nogit.txt; \
+    grep git+https:// requirements.txt > /tmp/requirements-git.txt; \
+    pip install -r /tmp/requirements-nogit.txt && pip install -r /tmp/requirements-git.txt; fi && \
+    rm /tmp/requirements-nogit.txt /tmp/requirements-git.txt
```

Comment:
Although I understand why this is necessary, it's a bit hard to read. I want to suggest to pipe the outcome of `grep` to `pip install -r /dev/stdin`, this would save a few lines since you don't have to deal with the different files.